### PR TITLE
Re-enable wasm32-unknown-emscripten target.

### LINKED ISSUE #839: Emscripten Targets Fail with Container Images
The container images are currently disabled on CI, however, they can be built. Unfortunately, attempting to do so produces:

```bash
error: linking with `emcc` failed: exit status: 1
  |
  = note: "emcc" "-s" "EXPORTED_FUNCTIONS=[\"_main\",\"_rust_eh_personality\"]" "/target/asmjs-unknown-emscripten/debug/deps/hellopp-4c6a3d6639ab24e0.1hwx1m4z77xgklfq.rcgu.o" "/target/asmjs-unknown-emscripten/debug/deps/hellopp-4c6a3d6639ab24e0.3olwem368bbcwnj6.rcgu.o" "/target/asmjs-unknown-emscripten/debug/deps/hellopp-4c6a3d6639ab24e0.42n3za5etetqf8wx.rcgu.o" "/target/asmjs-unknown-emscripten/debug/deps/hellopp-4c6a3d6639ab24e0.4bj48wwofmo2wxg5.rcgu.o" "/target/asmjs-unknown-emscripten/debug/deps/hellopp-4c6a3d6639ab24e0.4jy695kpqag0b4sv.rcgu.o" "/target/asmjs-unknown-emscripten/debug/deps/hellopp-4c6a3d6639ab24e0.kbgciut9opgn29.rcgu.o" "/target/asmjs-unknown-emscripten/debug/deps/hellopp-4c6a3d6639ab24e0.weu0i7ngo91mteq.rcgu.o" "/target/asmjs-unknown-emscripten/debug/deps/hellopp-4c6a3d6639ab24e0.2i5frysyjwi3f262.rcgu.o" "-L" "/target/asmjs-unknown-emscripten/debug/deps" "-L" "/target/debug/deps" "-L" "/target/asmjs-unknown-emscripten/debug/build/hellopp-b9600f35a3c2512d/out" "-L" "/rust/lib/rustlib/asmjs-unknown-emscripten/lib" "-l" "hellopp" "-l" "stdc++" "/rust/lib/rustlib/asmjs-unknown-emscripten/lib/libstd-fb2df19118db310d.rlib" "/rust/lib/rustlib/asmjs-unknown-emscripten/lib/libpanic_unwind-8386994b3a04ec77.rlib" "/rust/lib/rustlib/asmjs-unknown-emscripten/lib/libstd_detect-a104798db2201fc8.rlib" "/rust/lib/rustlib/asmjs-unknown-emscripten/lib/librustc_demangle-757cd8e06e392eda.rlib" "/rust/lib/rustlib/asmjs-unknown-emscripten/lib/libhashbrown-2c8bb84b50611da0.rlib" "/rust/lib/rustlib/asmjs-unknown-emscripten/lib/librustc_std_workspace_alloc-4a1eb87aaa56a6c1.rlib" "/rust/lib/rustlib/asmjs-unknown-emscripten/lib/libunwind-83c302b3cc8b7dbd.rlib" "/rust/lib/rustlib/asmjs-unknown-emscripten/lib/libcfg_if-9ada802475829674.rlib" "/rust/lib/rustlib/asmjs-unknown-emscripten/lib/liblibc-f9e2f3b41c6b1d5a.rlib" "/rust/lib/rustlib/asmjs-unknown-emscripten/lib/liballoc-ecb78a21f9518a04.rlib" "/rust/lib/rustlib/asmjs-unknown-emscripten/lib/librustc_std_workspace_core-58e0792c29686b11.rlib" "/rust/lib/rustlib/asmjs-unknown-emscripten/lib/libcore-ebe70f38647fc4a3.rlib" "/rust/lib/rustlib/asmjs-unknown-emscripten/lib/libcompiler_builtins-2c31a05cb537eb0d.rlib" "-l" "c" "-s" "DISABLE_EXCEPTION_CATCHING=0" "-L" "/rust/lib/rustlib/asmjs-unknown-emscripten/lib" "-L" "/rust/lib/rustlib/asmjs-unknown-emscripten/lib/self-contained" "-o" "/target/asmjs-unknown-emscripten/debug/deps/hellopp-4c6a3d6639ab24e0.js" "-O0" "--memory-init-file" "0" "-g4" "-s" "DEFAULT_LIBRARY_FUNCS_TO_INCLUDE=[]" "-s" "ERROR_ON_UNDEFINED_SYMBOLS=1" "-s" "ASSERTIONS=1" "-s" "ABORTING_MALLOC=0" "-Wl,--fatal-warnings" "-s" "WASM=0"
  = note: emcc: warning: please replace -g4 with -gsource-map [-Wdeprecated]
          emcc: error: wasm2js does not support source maps yet (debug in wasm for now)
```

### PR DESCRIPTION
**Pull request by Alexhuszagh**
Uses a trick described in rust-lang/rust/issues#85821#issuecomment-969369677. Note that this still allows it to run, despite there seemingly being no entry point: this has no effect on the resulting code.

Related to #609. Partially fixes #839.

### CODE REVIEW

**Comment by Alexhuszagh** [comment]
Note that `asmjs-unknown-emscripten` cannot be re-enabled, due to issues described in the updated CI action and in #839. I've tested it against `stdexcept` in a C++ target, which seemingly was the issue before, and it works fine. The file compiles, but when run properly complains that Rust cannot catch foreign exceptions:

```cpp
#include <iostream>
#include <stdexcept>

extern "C" {
  void hello() {
    std::cout << "Hello, world!" << std::endl;
     throw std::invalid_argument( "received negative value" );
  }
}
```

**Comment by Alexhuszagh** [comment]
bors try --target wasm32-unknown-emscripten

File: CHANGELOG.md [hunk 1]
```diff
@@ -31,6 +31,7 @@ This project adheres to [Semantic Versioning](http://semver.org/).
 
 ### Changed
 
+- #840 - re-enabled `wasm32-unknown-emscripten` target.
```
Comment (Emilgardis) [inline CHANGELOG.md:34]: remove, this is not a change from 0.2.1

**Review by Emilgardis — APPROVED** [review]
should be added to readme too, but otherwise, nice catch!

**Comment by Alexhuszagh** [comment]
I've updated the README and also updated the extract target info for this, since I also forgot to do this for solaris. Feel free to merge this whenever ready.

**Comment by Jules-Bertholet** [comment]
The upstream Rust issue has since been resolved (https://github.com/rust-lang/rust/issues/8582#issuecomment-969369677), and Emscripten has also made a change to support older Rust versions again (https://github.com/emscripten-core/emscripten/pull/17222). The emscripten fix has made it to the latest release (https://hub.docker.com/layers/emsdk/emscripten/emsdk/3.1.14/images/sha256-4c3e0a0dac61430b719e82118ae9b2c7480902a2713267e80fa296d39f7ab921). So this PR should probably be reverted in favor of a bump to the emscripten version used in the Dockerfile

**Pull request merged**